src/site/content/en/blog/serial/index.md

```diff
@@ -253,6 +253,94 @@ while (true) {
 }
 ```
 
+To specify the buffer into which data is read instead of the stream allocating a buffer for each chunk, get a [ReadableStreamBYOBReader] by calling `port.readable.getReader({ mode: "byob" })` and read data into a new `ArrayBuffer`. Note that the Web Serial API supports "Bring Your Own Buffer" readers in Chrome 106 or later.
+
+```js
+try {
+  const reader = port.readable.getReader({ mode: "byob" });
+  let buffer = new ArrayBuffer(512);
+  // Call reader.read() to read data into the buffer...
+} catch (error) {
+  if (error instanceof TypeError) {
+    // BYOB readers are not supported.
+    // Fallback to port.readable.getReader()...
+  }
+}
+```
+
+Here's an example of how to reuse the buffer out of `value.buffer`:
+
+```js
+const bufferSize = 8 * 1024; // 8kB
```

Comment:
Another potential footgun, if the developer is providing a large buffer to `read()` they should make sure to set `bufferSize` on `open()` to at least as much.

Reply:
I've updated one example to include that hint.

Reply:
Can you put that in this example? It's not important for the "read exactly this much" example because in that case we retry until the whole buffer is filled. In the case where we are explicitly trying to reuse the buffer from offset 0 however there's no point specifying a large buffer if the buffer size specified on open is smaller.

Reply:
Done. Sorry for that.